public/apps/configuration/panels/internal-user-edit/internal-user-edit.tsx

```diff
@@ -117,12 +126,23 @@ export function InternalUserEdit(props: InternalUserEditDeps) {
         updateObject.password = password;
       }
 
-      await updateUser(props.coreStart.http, userName, updateObject);
+      await updateUser(
+        props.coreStart.http,
+        userName,
+        updateObject,
+        createDataSourceQuery(dataSource.id)
+      );
 
       setCrossPageToast(buildUrl(ResourceType.users), {
         id: 'updateUserSucceeded',
         color: 'success',
-        title: getSuccessToastMessage('User', props.action, userName),
+        title: `${getSuccessToastMessage(
+          'User',
+          props.action,
+          userName,
+          dataSourceEnabled,
```

Comment:
What is the difference in the use between `dataSourceEnabled` and `dataSource` here? Are both necessary?

Reply:
The main reason for these two variables is that there are a few edge cases to consider. If the feature overall is disabled, we should not show any information about it, even if it is null or has data. If the feature overall is enabled, we want to show information about it, even if the id is empty (i.e. local cluster). Additionally, we also have to think about cases in which they may have the datasource in the URL, but disable the feature. Then, even if they have a datasource in the URL we shouldn't show anything (which wouldn't be possible without the boolean, since we try to read from the URL to support link sharing and auto logout/login feature).